apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SessionPool.cs

```diff
@@ -129,18 +129,26 @@ private void MaintainPool()
         /// </summary>
         /// <returns>A snapshot of statistics for this pool.</returns>
         public Statistics GetStatisticsSnapshot() =>
-            new Statistics(_targetedPools.ToArray().Select(p => p.Value.GetStatisticsSnapshot()).ToList().AsReadOnly());
-
+            new Statistics(_targetedPools.ToArray().Select(p => p.Value.GetStatisticsSnapshot()).ToList().AsReadOnly());
+
         /// <summary>
         /// Provides a snapshot of statistics for a database-specific pool.
         /// </summary>
         /// <returns>A snapshot of statistics for this pool.</returns>
-        public DatabaseStatistics GetStatisticsSnapshot(DatabaseName databaseName)
-        {
-            GaxPreconditions.CheckNotNull(databaseName, nameof(databaseName));
-            return _targetedPools.TryGetValue(databaseName, out var pool) ? pool.GetStatisticsSnapshot() : null;
+        [Obsolete("Use its overloaded method with SessionPoolKey instead.", error: false)]
+        public DatabaseStatistics GetStatisticsSnapshot(DatabaseName databaseName)

```

Comment:
I think we should not deprecate methods like AcquireSessionAsync, WhenPoolReady, As they are still useful for users who are not using FGAC. They don't need to worry about what the SessionPoolKey is and all of those things. What's your thoughts on the same.

Reply:
Yes, I think I agree, what I would like is to be able to have a `SessionPoolKeyStatistics GetStatisticsSnapshot(DatabaseName name)` which we cannot because we have the obsolete one.

Maybe the new ones should be called `GetPoolStatisticsSnapshot` and then we could have one receiving the `SessionPoolKey` and another one receiving just the `DatabaseName` and both returning `SessionPoolKeyStatistics`.